### ISSUE TITLE
Conversion between html and markdown in Rust with whitespace

### ISSUE THREAD

**Issue by alunturner** [matrix-org/matrix-rich-text-editor#407]
We currently have a slight issue with converting html to markdown in the rust model. Issue is as follows:

- html can support whitespace inside tags eg `<em>hello </em>` is valid html and if a user were to click italic, type 'hello ' and then click italic again to disable the italic mode, this is the html they would have input
- when we convert this to markdown, the output is `_hello _`
    - this is not valid markdown due to the space before the final underscore
    - this means that if we try to parse that markdown back into html we get the output `_hello _`

In summary, converting html => markdown => html at the moment is not perfect due to the difference in how the formats handle whitespace at the end of a 'node'.
